examples/nlp/language_modeling/megatron_gpt_validate.py

```diff
@@ -0,0 +1,136 @@
+)
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.app_state import AppState
+from nemo.utils.model_utils import inject_model_parallel_rank
+
+""" Example script showing how to run validation on a MegatronGPT model.
+
+    Sample usage:
+
+    From nemo model:
+
+    python megatron_gpt_validate.py \
+        trainer.devices=4 \
+        trainer.num_nodes=1 \
+        trainer.limit_val_batches=10 \
+        trainer.max_steps=100 \
+        tensor_model_parallel_size=1 \
+        pipeline_model_parallel_size=4 \
+        trainer.precision=bf16 \
+        gpt_model_file=/path/to/megatron_gpt_tp_1_pp4.nemo
+    
+    from PTL checkpoint:
+    python megatron_gpt_validate.py \
+        trainer.devices=4 \
+        trainer.num_nodes=1 \
+        trainer.limit_val_batches=10 \
+        trainer.max_steps=100 \
+        tensor_model_parallel_size=1 \
+        pipeline_model_parallel_size=4 \
+        virtual_pipeline_model_parallel_size=4 \
+        trainer.precision=bf16 \
+        checkpoint_dir='/path/to/experiment/checkpoints' \
+        checkpoint_name='megatron_gpt--val_loss=7.78-step=100-consumed_samples=6336.0-last.ckpt' \
+        hparams_file='/path/to/experiment/hparams.yaml
+
+"""
+
+
+@hydra_runner(config_path="conf", config_name="megatron_gpt_validate_config")
+def main(cfg) -> None:
+
+    trainer = Trainer(strategy=NLPDDPStrategy(), **cfg.trainer)
+
+    assert (
+        cfg.trainer.devices * cfg.trainer.num_nodes
+        == cfg.tensor_model_parallel_size * cfg.pipeline_model_parallel_size
+    ), "devices * num_nodes should equal tensor_model_parallel_size * pipeline_model_parallel_size"
+
+    if cfg.gpt_model_file:
+        logging.info(f"Restoring model from {cfg.gpt_model_file}")
+        save_restore_connector = NLPSaveRestoreConnector()
+        if os.path.isdir(cfg.gpt_model_file):
+            save_restore_connector.model_extracted_dir = cfg.gpt_model_file
+
+        pretrained_cfg = MegatronGPTModel.restore_from(
+            restore_path=cfg.gpt_model_file,
+            trainer=trainer,
+            return_config=True,
+            save_restore_connector=save_restore_connector,
+        )
+        OmegaConf.set_struct(pretrained_cfg, True)
+        with open_dict(pretrained_cfg):
+            pretrained_cfg.sequence_parallel = False
+            pretrained_cfg.activations_checkpoint_granularity = None
+            pretrained_cfg.activations_checkpoint_method = None
+            pretrained_cfg.precision = trainer.precision
+            if trainer.precision == "16":
+                pretrained_cfg.megatron_amp_O2 = False
+        model = MegatronGPTModel.restore_from(
+            restore_path=cfg.gpt_model_file,
+            trainer=trainer,
+            override_config_path=pretrained_cfg,
+            save_restore_connector=save_restore_connector,
+            map_location=f'cuda:{trainer.local_rank}',  # map_location is needed for converted models
```

Comment:
I don't use this usually, is it needed ?

Reply:
only for converted checkpoints

Reply:
Right I do inference with converted checkpoint and had no issues. Oh well if it doesn't hurt then it's ok

Reply:
I'll test this again to see if we do need it

Reply:
lets merge this in for current release and we can remove the map location if we don't need it later - doesn't hurt keeping it there for now